include/envoy/common/platform.h

```diff
@@ -19,13 +19,23 @@
 #include <mswsock.h>
 #include <ws2tcpip.h>
 
+#undef NTDDI_VERSION
+#define NTDDI_VERSION WDK_NTDDI_VERSION
+
+#include <mstcpip.h>
+
+#ifndef SIO_TCP_INFO
+static_assert(false, "SIO_TCP_INFO not defined");
```

Comment:
Do we know when exactly this was introduced? If it was roughly around the time afunix.h was introduced, do we need a similar comment as the afunix one below?

Reply:
It is a bit different than `<afunix.h>` since with `afunix.h` the whole header will be missing while here only one definition in missing. In the later case we can add a static assert.

When we fix the version issue I can add a static assert. So the logic will go as follows:

1. include `#include <mstcpip.h>`
2. If your version of the sdk is old then `SIO_TCP_INFO` will not be defined since the definition is guarded by:
```#if (NTDDI_VERSION >= NTDDI_WIN10_RS2)
#define SIO_TCP_INFO                        _WSAIORW(IOC_VENDOR,39)
#endif // NTDDI_VERSION >= NTDDI_WIN10_RS2
```
3. The `static_assert(false, "SIO_TCP_INFO not defined");` will be hit with a nice error message.

Reply:
For this one I will leave the comment as pending until we figure out how the headers order should work properly and then I will add it as a static assert.

Reply:
Based on the help of @nigriMSFT offline, the previous definition was at `envoy_internal.bzl` which was for an older windows version. I have bumped it to windows rs4 which according to the sources [build to version mapping](https://support.lenovo.com/ca/en/solutions/ht502786) and [af_unix announcement](https://devblogs.microsoft.com/commandline/af_unix-comes-to-windows/) is the minimum version that we need.